### PR TITLE
fix: remove the dependency on protobuf-lite for tombstones

### PR DESCRIPTION
**Pull request by supervacuus**
## :scroll: Description

As the title says, it replaces the dependency on a huge generic protobuf decoder with a tiny tombstone-specific package called `epitaph` ([GH](https://github.com/abovevacant/epitaph), [central](https://central.sonatype.com/artifact/com.abovevacant/epitaph)). The dependency is MIT licensed and can be incorporated into the Sentry ecosystem in whichever ways make sense down the road.

## :bulb: Motivation and Context

Part of the wrap-up milestone in https://linear.app/getsentry/project/tombstone-support-android-0024cb6e3ffd

* eliminates conflicts with client dependencies (https://github.com/getsentry/sentry-java/issues/5039, https://github.com/getsentry/sentry-java/pull/5044)
* reduces the size of the release AAR by 25%; 371 KiB instead 495 KiB (compared with [8.34.0](https://github.com/getsentry/sentry-java/releases/tag/8.34.0))
* since the tombstone protobuf format changes very infrequently (many years between changes), having a low dependency direct decoder is a sensible trade-off for a mobile SDK.

https://linear.app/getsentry/issue/ANDROID-267/switch-to-dependency-free-protobuf-runtime

## :green_heart: How did you test it?

Tested the same way as the dependency it replaces. The library, though, has a battery of test cases for encoding errors and edge cases.

## :pencil: Checklist

- [x] I added GH Issue ID _&_ Linear ID
- [x] I added tests to verify the changes.
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [ ] I updated the wizard if needed.
- [ ] Review from the native team if needed.
- [ ] No breaking change or entry added to the changelog.
- [ ] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: sentry-android-core/src/main/java/io/sentry/android/core/internal/tombstone/TombstoneParser.java [hunk 1]
```diff
@@ -27,7 +34,7 @@
 
 public class TombstoneParser implements Closeable {
 
-  private final InputStream tombstoneStream;
+  @Nullable private final InputStream tombstoneStream;
```
[the remark this answers is not in the document]
Comment (supervacuus) [reply sentry-android-core/src/main/java/io/sentry/android/core/internal/tombstone/TombstoneParser.java:37]: I was sure this would come from the bots :-D

Yes, theoretically true; in production, the `InputStream` will never be `null`, because that ctor is only used for tests (so we can construct test edge-cases via builder without having to go down and encode test tombstones into a byte-array stream, as we did for `protobuf-lite`).

But yeah, let's gate it.

File: sentry-android-core/src/main/java/io/sentry/android/core/internal/tombstone/TombstoneParser.java [hunk 2]
```diff
@@ -313,7 +329,7 @@ private DebugMeta createDebugMeta(@NonNull final TombstoneProtos.Tombstone tombs
         currentModule = new ModuleAccumulator(mapping);
       } else if (currentModule != null && mappingName.equals(currentModule.mappingName)) {
         // Extend the current module with this mapping (same file, continuation)
-        currentModule.extendTo(mapping.getEndAddress());
+        currentModule.extendTo(mapping.endAddress);
```
[the remark this answers is not in the document]
Comment (supervacuus) [reply sentry-android-core/src/main/java/io/sentry/android/core/internal/tombstone/TombstoneParser.java:332]: Same as above: https://github.com/getsentry/sentry-java/pull/5157#discussion_r2889221556
